File: datasets_v2/finance/gnucash/swiftui/Dialog_transfer_funds.swift
import SwiftUI

struct Dialog_transfer_funds: View {
  var body: some View {
    ZStack {
      ScrollView {
        VStack(alignment: .leading, spacing: 0) {
          HStack(alignment: .center, spacing: 0) {
            Text("Amount").font(.system(size: 16)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("$ 2000.00").font(.system(size: 18)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
          }
          .padding(10.0)
          HStack(alignment: .center, spacing: 0) {
            Text("From:").font(.system(size: 16)).multilineTextAlignment(.leading).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("USD").font(.system(size: 18)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("To:").font(.system(size: 16)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("EUR").font(.system(size: 18)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
          }
          .padding(10.0)
          Text("Provide either the converted amount or exchange rate in order to transfer funds").frame(maxWidth: .infinity, alignment: .leading).padding(10.0)
          HStack(alignment: .center, spacing: 0) {
            /* TODO: RadioButton -> Toggle fallback */
            Toggle("Option", isOn: .constant(false)).frame(maxWidth: .infinity)
            VStack(alignment: .leading, spacing: 8) {
              /* TODO: TextInputLayout label */
              TextField("Exchange rate", text: .constant("")).frame(maxWidth: .infinity)
            }
            .frame(maxWidth: .infinity)
          }
          .padding(.top, 10.0)
          Text("1 USD = 1.34 EUR").multilineTextAlignment(.center).frame(maxWidth: .infinity, alignment: .center).padding(.bottom, 10.0)
          HStack(alignment: .center, spacing: 0) {
            /* TODO: RadioButton -> Toggle fallback */
            Toggle("Option", isOn: .constant(false)).frame(maxWidth: .infinity)
            VStack(alignment: .leading, spacing: 8) {
              /* TODO: TextInputLayout label */
              TextField("Converted Amount", text: .constant("")).frame(maxWidth: .infinity)
            }
            .frame(maxWidth: .infinity)
            Text("EUR").font(.system(size: 16)).multilineTextAlignment(.center).frame(maxWidth: .infinity)
          }
          .padding(10.0)
          VStack(alignment: .leading, spacing: 0) {
            Spacer()
            HStack(alignment: .center, spacing: 0) {
              Button(action: { }) {
                Text("Cancel").foregroundColor(Color(red: 1, green: 0.6705882352941176, blue: 0, opacity: 1))
              }
              .disabled(false)
              .buttonStyle(.plain)
              .frame(maxWidth: .infinity)
              Button(action: { }) {
                Text("Save").foregroundColor(Color(red: 1, green: 0.6705882352941176, blue: 0, opacity: 1))
              }
              .disabled(false)
              .buttonStyle(.plain)
              .frame(maxWidth: .infinity)
            }
            .frame(maxWidth: .infinity)
          }
          .frame(maxWidth: .infinity, maxHeight: .infinity)
          .padding(.top, 5.0)
        }
        .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Dialog_transfer_funds_Previews: PreviewProvider {
  static var previews: some View {
    Dialog_transfer_funds()
  }
}

File: experiments/finance/gnucash/compose/dialog_transfer_funds.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun dialog_transfer_funds() {
  Box(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
    Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().verticalScroll(rememberScrollState())) {
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
        Row(modifier = Modifier.padding(10.0.dp)) {
          Text(text = "Amount", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 16.sp)
          Text(text = "$ 2000.00", modifier = Modifier.wrapContentHeight().weight(2.0f), fontSize = 18.sp)
        }
        Row(modifier = Modifier.padding(10.0.dp)) {
          Text(text = "From:", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 16.sp, textAlign = TextAlign.Start)
          Text(text = "USD", modifier = Modifier.wrapContentHeight().weight(1.5f), fontSize = 18.sp)
          Text(text = "To:", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 16.sp)
          Text(text = "EUR", modifier = Modifier.wrapContentHeight().weight(1.5f), fontSize = 18.sp)
        }
        Text(text = "Provide either the converted amount or exchange rate in order to transfer funds", modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(10.0.dp))
        Row(modifier = Modifier.padding(top = 10.0.dp)) {
          Row(modifier = Modifier.wrapContentHeight().weight(0.5f)) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            /* TODO label */
          }
          Column(modifier = Modifier.wrapContentHeight().weight(2.0f)) {
            
            OutlinedTextField(value = "", onValueChange = {}, placeholder = { Text("Exchange rate") }, modifier = Modifier.fillMaxWidth().wrapContentHeight())
          }
        }
        Text(text = "1 USD = 1.34 EUR", modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(bottom = 10.0.dp), textAlign = TextAlign.Center)
        Row(modifier = Modifier.padding(10.0.dp)) {
          Row(modifier = Modifier.wrapContentHeight().weight(0.5f)) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            /* TODO label */
          }
          Column(modifier = Modifier.wrapContentHeight().weight(2.0f)) {
            
            OutlinedTextField(value = "", onValueChange = {}, placeholder = { Text("Converted Amount") }, modifier = Modifier.fillMaxWidth().wrapContentHeight())
          }
          Text(text = "EUR", modifier = Modifier.wrapContentWidth().wrapContentHeight().weight(0.7f), fontSize = 16.sp, textAlign = TextAlign.Center)
        }
        Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().padding(top = 5.0.dp)) {
          Spacer(modifier = Modifier.weight(1f))
          Row(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
            Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
              Text(text = "Cancel", color = Color(0xFFFFAB00))
            }
            Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
              Text(text = "Save", color = Color(0xFFFFAB00))
            }
          }
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun dialog_transfer_funds_preview() {
    MyApplicationTheme {
        dialog_transfer_funds()
    }
}
